update capacitor version to version 7

alpay6767 opened the pull request:
I updated the capacitor version to version 7 so i could use this framework in my capacitor 7 project. This works perfectly for me now.

alpay6767 committed:
update version to version 7

juanaza commented:
Hey @alpay6767 why are you using this plugin instead of https://github.com/robsonos/nordic-dfu ?

I am having the same issue with the plugin I mentioned not being updated to capacitor 7

alpay6767 commented:
i tried the other plugin you mentioned but when i updated it to support capacitor 7 it showed me some errors. So i used this project, forked it and updated the version. This worked perfectly fine for me 👍

juanaza commented:
Hey @alpay6767 I ended up installing your forked capacitor 7 version and tested on both IOS 18 and Android 14 and they are running great.

Two things I would like to comment and get your opinion:
- In order to work on Android 14: i had to declare <service android:name="com.arklnk.plugins.dfu.DfuService" android:foregroundServiceType="connectedDevice" /> inside the <application> tag in the AndroidManifest.xml  and I had also to add     <uses-permission android:name="android.permission.FOREGROUND_SERVICE_CONNECTED_DEVICE" />. This prevented from getting a fatal exception that was closing the APP.
- Are you planning on releasing your updated capacitor 7 version on NPM? since the owner (https://www.npmjs.com/package/capacitor-nordic-dfu) is not updating it?
